src/Type/Equality.purs

```diff
@@ -0,0 +1,35 @@
+module Type.Equality
+  ( class TypeEquals
+  , to
+  , from
+  , class TypeEqualsBool
+  ) where
+
+import Type.Data.Boolean (kind Boolean, True, False)
+
+-- | This type class asserts that types `a` and `b`
+-- | are equal.
+-- |
+-- | The functional dependencies and the single
+-- | instance below will force the two type arguments
+-- | to unify when either one is known.
+-- |
+-- | Note: any instance will necessarily overlap with
+-- | `refl` below, so instances of this class should
+-- | not be defined in libraries.
+class TypeEquals a b | a -> b, b -> a where
+  to :: a -> b
+  from :: b -> a
+
+instance refl :: TypeEquals a a where
+  to a = a
+  from a = a
+
+-- | This type class asserts that types `a` and `b`
+-- | are or are not equal, depending on the type of `o`.
```

Comment:
I think I'd expect this description to also describe the information flow declared by the functional dependencies, perhaps include something like:
```
Decide if types `a` and `b` are equal.  `o` will be either `True`, if `a` is `b`, or `False` otherwise.
When `o` is specified to be `True`/`False` then this type class acts as an assertion that `a` and `b` are equal/not equal, respectively.
```
Don't necessarily take this verbatim, if you can think of something better.

Reply:
Thanks! Took it mostly verbatim :)